lib/pay/payments.ex

```diff
@@ -780,41 +782,40 @@ defmodule Pay.Payments do
     |> Repo.insert()
   end
 
+
+
   @doc """
-  Updates a gateway_account_card_type.
+  Deletes a GatewayAccountCardType.
 
   ## Examples
 
-      iex> update_gateway_account_card_type(gateway_account_card_type, %{field: new_value})
+      iex> delete_gateway_account_card_type(gateway_account_card_type)
       {:ok, %GatewayAccountCardType{}}
 
-      iex> update_gateway_account_card_type(gateway_account_card_type, %{field: bad_value})
+      iex> delete_gateway_account_card_type(gateway_account_card_type)
       {:error, %Ecto.Changeset{}}
 
   """
-  def update_gateway_account_card_type(
-        %GatewayAccountCardType{} = gateway_account_card_type,
-        attrs
-      ) do
-    gateway_account_card_type
-    |> GatewayAccountCardType.changeset(attrs)
-    |> Repo.update()
+  def delete_gateway_account_card_type(%GatewayAccountCardType{} = gateway_account_card_type) do
+    Repo.delete(gateway_account_card_type)
   end
 
+
+  #TODO : Replace Repo.delete_all with put_assoc
```

Comment:
might be worth adding a why to the TODO note. it's not obvious why we need to do it and may not be obvious in a few months.